packages/svelte/src/runtime/internal/Component.js

```diff
@@ -125,8 +128,9 @@ export function init(
 		? instance(component, options.props || {}, (i, ret, ...rest) => {
 				const value = rest.length ? rest[0] : ret;
 				if ($$.ctx && not_equal($$.ctx[i], ($$.ctx[i] = value))) {
-					if (!$$.skip_bound && $$.bound[i]) $$.bound[i](value);
+					if (!$$.skip_bound && is_function($$.bound[i])) $$.bound[i](value);
 					if (ready) make_dirty(component, i);
+					else $$.bound[i] = true; // dirty flag consumed in bind()
```

Comment:
The `$$invalidate` runtime function, when `ready == false` sets `$$.bound[i] = true` in order to cater for case 3. This mark is essentially the same as `$$.dirty` which is not available before `ready`.